src/CoreArbiterServer.cc

```diff
@@ -753,7 +753,11 @@ CoreArbiterServer::timeoutThreadPreemption(int timerFd)
 {
     if (!testingSkipSocketCommunication) {
         uint64_t time;
-        read(timerFd, &time, sizeof(uint64_t));
+        ssize_t ret = read(timerFd, &time, sizeof(uint64_t));
+        if (ret == -1) {
+            LOG(ERROR, "Error reading time: %s", strerror(errno));
+            return;
```

Comment:
Given that the above `read` should effectively never fail, it's probably reasonable to call `exit(1)` here, since a failure here indicates something seriously unexpected.